src/main/java/net/openhft/chronicle/queue/impl/single/TableDirectoryListingReadOnly.java

```diff
@@ -38,7 +39,18 @@ protected void checkReadOnly(@NotNull TableStore<?> tableStore) {
     public void init() {
         throwExceptionIfClosedInSetter();
 
-        initLongValues();
+        // it is possible if r/o queue created at same time as r/w queue for longValues to be only half-written
+        final long timeoutMillis = System.currentTimeMillis() + 500;
+        while (true) {
+            try {
+                initLongValues();
+                break;
+            } catch (Exception e) {
+                if (System.currentTimeMillis() > timeoutMillis)
+                    throw e;
+                Jvm.pause(1);
```

Comment:
If there's an error here I think we need to close any of the LongValues that were successfully acquired first, or we'll be leaving them open when they're re-assigned on the next iteration

Reply:
Or if read-only were a distinct implementation, it's initLongValues could only try and acquire values it hasn't already acquired ;)

Reply:
Problem is, the LongValues won't get assigned to a variable if an exception is thrown, so closing them will only close the ones that were successfully acquired